File: c/resistor-color/resistor_color.c
#include "resistor_color.h"

uint16_t color_code(resistor_band_t color) {
    /* No need for cast thank to implicit conversion */
    return color;
}

static resistor_band_t colors_[] = {
    BLACK, BROWN, RED, ORANGE, YELLOW, GREEN, BLUE, VIOLET, GREY, WHITE
};

resistor_band_t * colors() {
    return &colors_[0];
}

File: lua/resistor-color/resistor-color.lua
return {
   color_code = function(color)
      local colors = {'black', 'brown', 'red', 'orange', 'yellow', 'green', 'blue', 'violet', 'grey', 'white'}
      for k, v in pairs(colors) do
         if v == color then
            return k - 1
         end
      end
   end
}
